src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -142,6 +142,53 @@ static ColumnWithTypeAndName readColumnWithStringData(std::shared_ptr<arrow::Chu
     return {std::move(internal_column), std::move(internal_type), column_name};
 }
 
+/// Inserts chars and offsets right into internal column data to reduce an overhead.
+/// Internal offsets are shifted by one to the right in comparison with Arrow ones. So the last offset should map to the end of all chars.
+/// Also internal strings are null terminated.
+static ColumnWithTypeAndName readColumnWithLargeStringData(std::shared_ptr<arrow::ChunkedArray> & arrow_column, const String & column_name)
```

Comment:
As I can see, this function differs with `readColumnWithStringData` in only one line where we cast chunk to `arrow::LargeBinaryArray` instead of `arrow::BinaryArray`. Let's remove code duplication and make one function with template type of arrow chunk

Reply:
I'm not sure how to do that in C++, but happy for someone to do this.